plugins/pack-deploy.yaml

```diff
@@ -0,0 +1,29 @@
+apiVersion: krew.googlecontainertools.github.com/v1alpha2
+kind: Plugin
+metadata:
+  name: pack-deploy
+spec:
+  version: "v0.1.0"
+  platforms:
+  - uri: https://github.com/dougsland/kubectl-pack-deploy/archive/kubectl-pack-deploy-0.1.0.tar.gz
+    sha256: 403b04620f47c8d67f09c0aa405723fc9acdf3cc98face60f4adb9ff8e358d7e
+    bin: kubectl-pack_deploy
+    files:
+    - from: "./*/kubectl-pack_deploy"
+      to: .
+    - from: "./*/LICENSE"
+      to: .
+    selector:
+      matchExpressions:
+      -   {key: os, operator: In, values: [darwin, linux]}
+  caveats: |
+    This plugin needs the following programs:
+    * kubectl neat plugin
+    Examples:
+    To generate the YAML file from a kibana deployment running in the cluster use:
+      kubectl pack-deploy kibana
+  description: |
+    Detects resources required from running deployments, statefulsets,
+    daemonsets and export it with all dependencies in a single YAML.
```

Comment:
From this description it is very hard to guess what the plugin is actually doing. IIUC, what it does is looking for resources whose names contain the given identifier. I think matching by name is a fragile way to export an application with multiple resources. You have to make it very clear how it works and avoid false promises.

Also, consider that the recommended way to select/group resources in k8s is via labels on the resources. Your plugin should default to that logic and only use the name matching as an explicit option.

Reply:
hey @corneliusweig thanks for the review. I am aware that we should move to labels. I just learned about kubectl plugins recently  and watched all videos in youtube from @ahmetb  and started doing a plugin (an here we are now) :).
Next version, we will move this to approach you mentioned.

Reply:
Unless you already have a solid idea and implementation in mind, I think this should be in krew-index yet. 

You can distribute this using custom indexes if you like.

Reply:
@ahmetb I'm a bit confused what you wanted to say. Maybe you forgot 'not'?

 Changing the mechanics of a plugin later on is much harder and may annoy your users. I suggest to make the decision now whether you want to use labels or names for selecting resources. I think that labels are much safer and better aligned with the paradigms of k8s, but ultimately it is your call.